sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/InputComponentStore.java

```diff
@@ -80,7 +83,7 @@ public InputComponentStore(DefaultInputModule root, AnalysisMode mode) {
     return inputFileCache.values().stream()
       .map(f -> (DefaultInputFile) f)
       .filter(DefaultInputFile::isPublished)
-      .filter(f -> !mode.isIncremental() || f.status() != Status.SAME)::iterator;
+      .filter(f -> (!mode.isIncremental() && !branchConfiguration.isShortLivingBranch()) || f.status() != Status.SAME)::iterator;
```

Comment:
if status is SAME, shouldn't we filter it regardless of the branch type (currently it should only happen for short living branches, right?)

Reply:
* "Normal analysis": write all components
* incremental / short living branches: write only changed components